android/titanium/src/java/org/appcelerator/titanium/io/TiFileProvider.java

```diff
@@ -24,14 +24,13 @@ public boolean onCreate() {
 		return true;
 	}
 
-	public static Uri createUriFrom(File file) {
+	public static Uri createUriFrom(String filePath) {
 		final TiApplication tiApp = TiApplication.getInstance();
 		if (tiApp == null) {
 			return null;
 		}
-
 		try {
-			String path = getUriPrefix() + file.getAbsolutePath();
+			String path = getUriPrefix() + filePath.replaceAll("^file://(/android_asset)?", "");
```

Comment:
When we wrote the `TiContentProvider`, we purposely limited the `createUriFrom()` method to only support `File` objects to limit it to only supporting external files on the filesystem. If we were to add support for reading asset files and resource files in the future (and we should), we would have to override the `ContentProvider.openAssetFile()` method in order to access embedded files within the APK as well change the URI prefix to better represent where the file came from (currently hardcoded to "/filesystem").

The test case for this code change is limited to external files. Can we keep the old way of doing it for now?

Reply:
It is still only limited to external files (even though there's regex that cleans our `/android_asset` identifier). I've just added the ability to pass a file path String in addition to the existing File object. Which is more appropriate for passing the `data` String when creating the `Intent` instead of needing to create a `File` object only to grab it's path again.